meshtastic-mqtt-connect.py

```diff
@@ -365,10 +372,27 @@ def publish_message(destination_id):
         mesh_packet.id = random.getrandbits(32)
         mesh_packet.to = destination_id
         mesh_packet.want_ack = True
-        # mesh_packet.want_ack = False
-        mesh_packet.channel = 0
+        mesh_packet.channel = 8
         mesh_packet.hop_limit = 3
 
+        if do_encrypt:
+            key_bytes = base64.b64decode(key.encode('ascii'))
+
+            nonce_packet_id = mesh_packet.id.to_bytes(8, "little")
+            nonce_from_node = node_number.to_bytes(8, "little")
+            # Put both parts into a single byte array.
+            nonce = nonce_packet_id + nonce_from_node
+
+            if key == "AQ==":
+                key_bytes = bytes([0xd4, 0xf1, 0xbb, 0x3a, 0x20, 0x29, 0x07, 0x59, 0xf0, 0xbc, 0xff, 0xab, 0xcf, 0x4e, 0x69, 0x01])
+
+            cipher = Cipher(algorithms.AES(key_bytes), modes.CTR(nonce), backend=default_backend())
+            encryptor = cipher.encryptor()
+            encrypted_bytes = encryptor.update(encoded_message.SerializeToString()) + encryptor.finalize()
+
+            mesh_packet.encrypted = encrypted_bytes
+            
```

Comment:
Need to make sure that the mesh_packed.decoded is set if you aren't going to encrypt, so add:

else:
    mesh_packet.decoded.CopyFrom(encoded_message)

Reply:
hmm, i put that before the if, on line 369
seems to work

Reply:
Also, this code needs to be removed at line 369 -- otherwise, even if encryption is on, it'll set the decoded packet and encryption will be ignored anyway!

Reply:
Actually, my comment may not be true, due to the field being a "oneof" -- so if decoded is set, and then encrypted is set, then encrypted should be the only thing actually set at the end of the function.  However, it is more readable and intentional to only set the decoded version in an "else" statement.

Reply:
TBH, I haven't tested this PR very well

Reply:
made some changes which I think are what you were suggesting
https://github.com/pdxlocations/Meshtastic-MQTT-Connect/commit/6ce2651ca6feb63ef386cbf23a623a8a2437ab03